File: scheduler-server/src/schemas/group.py
from pydantic import BaseModel, Field

from src.schemas.common import TimestampedModel


class GroupCreate(BaseModel):
    name: str = Field(min_length=1, max_length=120)
    description: str | None = None


class GroupRead(TimestampedModel):
    id: int
    name: str
    description: str | None


class GroupMemberAddItem(BaseModel):
    instance_id: int
    agent_id: int


class GroupMemberAddRequest(BaseModel):
    members: list[GroupMemberAddItem]


class GroupMemberRead(BaseModel):
    id: int
    group_id: int
    instance_id: int
    agent_id: int
    joined_at: str
    agent_key: str
    display_name: str
    role_name: str | None
    instance_name: str


class GroupDetail(BaseModel):
    id: int
    name: str
    description: str | None
    members: list[GroupMemberRead]

File: web-client/src/types/api/group.ts
export interface GroupResponse {
    id: number;
    name: string;
    description: string | null;
    created_at: string;
    updated_at: string;
}

export interface GroupMemberResponse {
    id: number;
    group_id: number;
    instance_id: number;
    agent_id: number;
    joined_at: string;
    agent_key: string;
    display_name: string;
    role_name: string | null;
    instance_name: string;
}

export interface GroupDetailResponse {
    id: number;
    name: string;
    description: string | null;
    members: GroupMemberResponse[];
}
